src/common/test-webserver.c

```diff
@@ -106,14 +108,19 @@ setup (TestFixture *fixture,
                                            test_case ? test_case->server_flags : COCKPIT_WEB_SERVER_NONE);
   g_clear_object (&cert);
 
+  if (test_case && test_case->forwarded_for_header)
+    cockpit_web_server_set_forwarded_for_header (fixture->web_server, test_case->forwarded_for_header);
   if (test_case && test_case->protocol_header)
     cockpit_web_server_set_protocol_header (fixture->web_server, test_case->protocol_header);
 
   /* We want to check all incoming requests to ensure that they match
    * our expectations about remote hostname and protocol.  Add a
    * "handler" that does that, but never claims to handle anything.
    */
-  fixture->expected_remote = g_strdup (address);
+  if (test_case && test_case->expected_remote)
+    fixture->expected_remote = g_strdup (test_case->expected_remote);
```

Comment:
Same thing as in the first commit; this is weird IMHO

Reply:
Ya, I'll clean that up.